### PR TITLE
Bug fix for negative reload intervals - bugz 1311459.

### PR DESCRIPTION
**Pull request by ramr**
@eparis / @smarterclayton alternative fix for #7589    PTAL Thx

### CODE REVIEW

File: pkg/cmd/infra/router/template.go [hunk 1]
```diff
@@ -139,6 +139,10 @@ func (o *TemplateRouterOptions) Complete() error {
 		o.StatsPort = statsPort
 	}
 
+	if nsecs := int(o.ReloadInterval.Seconds()); nsecs < 0 {
+		return fmt.Errorf("invalid reload interval: %v", nsecs)
```
Comment (smarterclayton) [inline pkg/cmd/infra/router/template.go]: Add "must be a positive duration" to help users figure out why.

**Comment by ramr** [comment]
@smarterclayton  done - updated the error message (squashed the commits).

**Comment by eparis** [comment]
[merge]

**Comment by smarterclayton** [comment]
Lgtm

**Pull request merged**